server/app/forms/MapQuestionForm.java

```diff
@@ -126,14 +141,26 @@ private void setFormWithQuestionSettings(ImmutableSet<QuestionSetting> settings)
         getSettingFromQuestionSettings(settings, MapSettingType.LOCATION_DETAILS_URL_GEO_JSON_KEY);
 
     this.filters = new ArrayList<>(getFiltersFromQuestionSettings(settings));
+    this.locationTag =
+        getSettingFromQuestionSettings(settings, MapSettingType.LOCATION_TAG_GEO_JSON_KEY);
   }
 
   /** Converts {@link QuestionSetting} back to form {@link Setting} for editing. */
   private Setting getSettingFromQuestionSettings(
       ImmutableSet<QuestionSetting> settings, MapSettingType type) {
     return settings.stream()
         .filter(setting -> setting.settingType().equals(type))
-        .map(setting -> new Setting(setting.settingValue(), ""))
+        .map(
+            setting ->
+                new Setting(
+                    setting.settingKey(),
+                    setting.settingValue().orElse(""),
+                    setting.localizedSettingDisplayName().isPresent()
+                        ? setting.localizedSettingDisplayName().get().getDefault()
+                        : "",
```

Comment:
(non-blocking, fyi) This is fine, but just want to show you a way to not need the ternary operator

```suggestion
                    setting.localizedSettingDisplayName().map(LocalizedStrings::getDefault).orElse(""),
```

Reply:
oho this is nice too